src/ds/logger.h

```diff
@@ -71,6 +74,39 @@ namespace logger
       static std::unique_ptr<ringbuffer::AbstractWriter> the_writer;
       return the_writer;
     }
+
+    // Count of milliseconds elapsed since enclave started, used to produce
+    // offsets to host time when logging from inside the enclave
+    static std::chrono::milliseconds ms;
+
+    static void tick(std::chrono::milliseconds ms_)
+    {
+      ms += ms_;
+    }
+
+    static std::chrono::milliseconds elapsed_ms()
+    {
+      return ms;
+    }
+#else
+    // Timestamp of first tick. Used by the host when receiving log messages
+    // from the enclave. Combined with the elapsed ms reported by the enclave,
+    // and used to compute the offset between time inside the enclave, and time
+    // on the host when the log message is received.
+    static ::timespec start;
+
+    static void set_start(
+      const std::chrono::time_point<std::chrono::system_clock>& start_)
+    {
+      start.tv_sec = std::chrono::time_point_cast<std::chrono::seconds>(start_)
+                       .time_since_epoch()
+                       .count();
+      start.tv_nsec =
+        std::chrono::time_point_cast<std::chrono::nanoseconds>(start_)
+          .time_since_epoch()
+          .count() -
+        start.tv_sec * 1000000000;
```

Comment:
Minor: Can we write this as `1'000'000'000`? I find it more readable, and much easier to confirm we have the correct number of 0s. In this case we use it often for a single purpose - it should perhaps be a named constant `ns_per_s`.

Reply:
Yes, absolutely!